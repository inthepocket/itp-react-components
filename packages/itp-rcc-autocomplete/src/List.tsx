import * as React from 'react';
import AutoCompleteContext from './AutoCompleteContext';
import ListItem, { ListItemInterface } from './ListItem';

const { Consumer } = AutoCompleteContext;

const List = ({
  getItemProps = () => false,
  getMenuProps = () => false,
  highlightedIndex,
  inputValue,
  items,
  selectedItem,
}: SuggestionListPropsInterface) => (
  <Consumer>
    {({ prefixCss, styles }) => {
      const className = styles ? styles.list : `${prefixCss}__list`;

      return (
        <ul {...getMenuProps()} className={className}>
          {items.map((item, index) => {
            const itemProps = getItemProps({
              index,
              item,
              key: item.value,
            });

            const isSelected =
              (selectedItem && selectedItem.value === item.value && selectedItem.value === inputValue) ||
              item.value === inputValue;

            return (
              <ListItem
                {...itemProps}
                id={item.value}
                isHighlighted={highlightedIndex === index}
                isSelected={isSelected}
                key={itemProps.key}
                value={item.value}
              />
            );
          })}
        </ul>
      );
    }}
  </Consumer>
);

interface SuggestionListPropsInterface {
  getItemProps?: Function;
  getMenuProps?: Function;
  highlightedIndex?: number | null;
  inputValue?: string | null;
  items: Array<ListItemInterface>;
  selectedItem?: ListItemInterface;
}

export default List;
