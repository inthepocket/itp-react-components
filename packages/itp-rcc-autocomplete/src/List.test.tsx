import * as React from 'react';
import { create } from 'react-test-renderer';
import List from './List';

const items = [
  { value: 'Amsterdam' },
  { value: 'Bangkok' },
  { value: 'Brussels' },
  { value: 'London' },
  { value: 'New York' },
  { value: 'Paris' },
];

test('Autocomplete List snapshot', () => {
  const tree = create((
    <List
      getItemProps={() => ({ key: Math.random() })}
      items={items}
    />
  ))
  .toJSON();

  expect(tree).toMatchSnapshot();
});