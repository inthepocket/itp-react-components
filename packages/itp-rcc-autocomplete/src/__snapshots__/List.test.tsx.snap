// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Autocomplete List snapshot 1`] = `
<ul
  className="card__list"
>
  <li
    className="card__listItem"
  >
    Amsterdam
  </li>
  <li
    className="card__listItem"
  >
    Bangkok
  </li>
  <li
    className="card__listItem"
  >
    Brussels
  </li>
  <li
    className="card__listItem"
  >
    London
  </li>
  <li
    className="card__listItem"
  >
    New York
  </li>
  <li
    className="card__listItem"
  >
    Paris
  </li>
</ul>
`;
